codechain/rpc_apis.rs

```diff
@@ -55,7 +55,7 @@ impl ApiDependencies {
     }
 }
 
-pub fn setup_rpc(mut handler: MetaIoHandler<()>) -> MetaIoHandler<()> {
+pub fn setup_rpc(mut handler: MetaIoHandler<(), impl Middleware<()>>) -> MetaIoHandler<(), impl Middleware<()>> {
```

Comment:
It means that the input and output can be different types which implement MiddleWare.
```rust
pub fn setup_rpc<M: MiddleWare<()>>(mut handler: MetaIoHandler<(), M>) -> MetaIoHandler<(), M> {
```

Reply:
I fixed it.